contrib/test_mergeBenchmarkSets/test_integration/__init__.py

```diff
@@ -0,0 +1,155 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import filecmp
+import os
+import subprocess
+import sys
+import unittest
+
+sys.dont_write_bytecode = True  # prevent creation of .pyc files
+
+here = os.path.relpath(os.path.dirname(__file__))
+base_dir = os.path.join(here, "..", "..", "..")
+merge_benchmark_sets = [
+    sys.executable,
+    os.path.join(base_dir, "contrib", "mergeBenchmarkSets.py"),
+]
+
+# Set to True to let tests overwrite the expected result with the actual result
+# instead of letting them fail.
+# Use this to update expected files if necessary. Do not commit this flag set to True!
+OVERWRITE_MODE = False
+
+
+class MergeBenchmarkSetsIntegrationTests(unittest.TestCase):
+    def run_cmd(self, *args):
+        try:
+            output = subprocess.check_output(
+                args=args, stderr=subprocess.STDOUT
+            ).decode()
+        except subprocess.CalledProcessError as e:
+            print(e.output.decode())
+            raise e
+        print(output)
+        return output
+
+    def merge_tables_and_compare(
+        self, resultfile, witness_files, no_overwrite=False, merge_suffix=""
+    ):
+        result = self.run_cmd(
+            *merge_benchmark_sets
+            + [resultfile]
+            + witness_files
+            + (["--no-overwrite-status-true"] if no_overwrite else [])
+        ).strip()
+        expected = os.path.join(
+            here,
+            "expected",
+            os.path.basename(resultfile) + ".merged" + merge_suffix + ".xml.bz2",
+        )
+        if OVERWRITE_MODE:
+            os.replace(result, expected)
+        else:
+            self.assert_content_equals(result, expected)
+            os.remove(result)  # TODO: More reliable cleanup
```

Comment:
How about using `tempfile.TemporaryDirectory` as a context manager (using `with`)? We actually prefer tests to not write somewhere inside the user's directories anyway.

Reply:
Had to add an option to `mergeBenchmarkSets` to be able to put the result files in the temp dir, but should be fine.